spyder/app/mainwindow.py

```diff
@@ -16,216 +16,191 @@
+# TODO: Is this order important or could be moved to the top?
+from qtpy import QtSvg  # Avoid Cannot mix incompatible Qt library on windows
+from qtpy import QtWebEngineWidgets # Avoid https://bugreports.qt.io/browse/QTBUG-46720
+
+# Local imports
+from spyder import requirements
+from spyder.api.translations import get_translation
+from spyder.app.cli_options import get_options
+# Slowest import is this one, try to optimize
+from spyder.app.utils import (qt_message_handler, set_opengl_implementation,
+                              setup_logging)
+from spyder.config.base import (DEV, get_conf_path, get_debug_level,
+                                get_image_path, reset_config_files,
+                                running_under_pytest)
+from spyder.config.manager import CONF
+from spyder.py3compat import PY3
+from spyder.utils.misc import getcwd_or_home
+from spyder.utils import encoding, programs
+from spyder.utils.qthelpers import qapplication
+from spyder.utils.windows import (set_attached_console_visible,
+                                  is_attached_console_visible,
+                                  set_windows_appusermodelid)
 
 
-#==============================================================================
-# Keeping a reference to the original sys.exit before patching it
-#==============================================================================
-ORIGINAL_SYS_EXIT = sys.exit
+# Logging
+# ----------------------------------------------------------------------------
+logger = logging.getLogger(__name__)
 
-#==============================================================================
-# Check requirements
-#==============================================================================
-from spyder import requirements
-requirements.check_path()
-requirements.check_qt()
-requirements.check_spyder_kernels()
 
-#==============================================================================
-# Windows only: support for hiding console window when started with python.exe
-#==============================================================================
-set_attached_console_visible = None
-is_attached_console_visible = None
-set_windows_appusermodelid = None
-if os.name == 'nt':
-    from spyder.utils.windows import (set_attached_console_visible,
-                                      is_attached_console_visible,
-                                      set_windows_appusermodelid)
+# Localization
+# ----------------------------------------------------------------------------
+_ = get_translation('spyder')
+
+
+# Constants
+# ----------------------------------------------------------------------------
+# Keep a reference to the splash screen, main application, icon sys.exit and
+# original working directory.
+APP_ICON = None
+CWD = None
+MAIN_APP = None
+ORIGINAL_SYS_EXIT = None
+SPLASH = None
+
+
+# Import helper methods
+# ----------------------------------------------------------------------------
+def import_qt_modules():

```

Comment:
Importing things in a function is highly non-standard, it has nothing to do with this PR and I don't like it. Please revert to the previous behavior and open a new PR about it if you're still interested in seeing it merged.